### ISSUE TITLE
How to create a Shared Project from the CLI?

### ISSUE THREAD

**Issue by raffaeler** [dotnet/sdk#4228]
Visual Studio provides the "Shared Project" template which is very useful to share source files across multiple "real" projects.
I can't see the equivalent in .NET Core 3 using `dotnet new --help`.
Did I miss something?
Will it be available in .NET 5?

**Comment by Phyyl** [comment]
I was also wondering about that. I'd love to see the shared project redone the way the csproj was.

**Comment by timeverts** [comment]
Yes. This would be extremely useful. It's a common project type in Visual Studio that isn't supported by the cli.

![image](https://github.com/dotnet/sdk/assets/15150632/bc65127c-aeee-4f00-bb13-e39a8f4bdf84)

Shared projects have been around [since Visual Studio 2015](https://www.c-sharpcorner.com/UploadFile/7ca517/shared-project-an-impressive-features-of-visual-studio-201/) and typically have a `*.projitems` and `*.shproj` file.

@joeloff, @v-wuzhai, any chance you could add support for these?
